### ISSUE TITLE
HI!

### ISSUE THREAD

**Issue by ghost** [aldro61/pu-learning#1]
Nifty tool. Have you considered adding a license to the project?

**Comment by aldro61** [comment]
Hi!

Thanks! Yes I have been thinking of adding one. I will look into this. Do you have any recommendations?

**Comment by ghost** [comment]
I've used BSD in the past. Basically means that everyone can do with it as the please. GPL means that you can do with it as you please, but if you release it to the public you have to include the modified source. I'd like to contribute once you have a license. :)

**Comment by aldro61** [comment]
I think BSD 3-clause would be a good choice! I was also thinking of integrating this code into the scikit-learn library. Would you still be willing to contribute if I forked sklearn and added the puAdapter class under sklearn.semi_supervised? I would rename it PosOnly, as this is the name that Elkan and Noto use in their article.

The current version of the puAdapter only integrates estimator 1. Although in their article Elkan and Noto propose 3 estimators. Providing a choice of estimator would be a nice improvement to the project.

The paper: http://cseweb.ucsd.edu/~elkan/posonly.pdf

**Comment by ghost** [comment]
PosOnly makes a lot of sense. semi_supervised also make sense, but I've also read papers that treat it as basically a supervised problem.

I agree providing a choice of estimators would be nice. Take a look at the call signature from sklearn.feature_selection.RFE. Something like that would be nice.

I have a sklearn neural net with dropout model sitting in my repo. I was thinking of rolling it into a separate installable module, so I could install it independently of updating sklearn itself. Otherwise you have to merge from upstream every time that it updates.

If you wanted to get it incorporated into the project you might open a feature enhancement issue on their page, including a fork of it.

I'd be love to contribute to it with you, even if it's just writing tests. After reading a little about PU classification, I'm starting to see it in a lot of datasets I work it.

**Comment by aldro61** [comment]
Hi JakeMick,

I just licensed the project under the BSD 3-clause license. I will look into including the PUAdapter to sklearn when I have a little bit more time. Thanks for your suggestions!
